File: bloc_de_recordatorios.java
import java.util.ArrayList;
import java.util.Scanner;
public class bloc_de_recordatorios {
    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);
        int opcion = 0;
        int cantidad = 0;
        ArrayList<String> titulo = new ArrayList<>();
        ArrayList<String> descripcion = new ArrayList<>();
        ArrayList<String> fecha = new ArrayList<>();
        ArrayList<String> hora = new ArrayList<>();
        ArrayList<String> prioridad = new ArrayList<>();
        ArrayList<String> categoria = new ArrayList<>();
        ArrayList<String> estado = new ArrayList<>();
        //---------------------------------------------//
        //--|menu_principal_de_bloc_de_recordatorios|--//
        //---------------------------------------------//
        while (opcion != 6) {
            System.out.println("===============================");
            System.out.println("   BLOC DE RECORDATORIOS");
            System.out.println("===============================");
            System.out.println("1. Crear recordatorio");
            System.out.println("2. Editar recordatorio");
            System.out.println("3. Eliminar recordatorio");
            System.out.println("4. Marcar como completado");
            System.out.println("5. Mostrar recordatorios");
            System.out.println("6. Salir");
            System.out.println("===============================");
            System.out.print("Seleccione una opcion: ");
            opcion = entrada.nextInt();
            entrada.nextLine();
            //---------------------------//
            //--|crear_el_recordatorio|--//
            //---------------------------//
            if (opcion == 1) {
                cantidad++;
                System.out.print("Ingrese el titulo: ");
                String nuevo_titulo = entrada.nextLine();
                System.out.print("Ingrese la descripcion: ");
                String nueva_descripcion = entrada.nextLine();
                System.out.print("Ingrese la fecha: ");
                String nueva_fecha = entrada.nextLine();
                System.out.print("Ingrese la hora: ");
                String nueva_hora = entrada.nextLine();
                System.out.println("Alta - Media - Baja");
                System.out.print("Ingrese la prioridad: ");
                String nueva_prioridad = entrada.nextLine();
                System.out.println("Trabajo - Estudio - Personal - Compras");
                System.out.print("Ingrese la categoria: ");
                String nueva_categoria = entrada.nextLine();
                String nuevo_estado = "Pendiente";
                titulo.add(nuevo_titulo);
                descripcion.add(nueva_descripcion);
                fecha.add(nueva_fecha);
                hora.add(nueva_hora);
                prioridad.add(nueva_prioridad);
                categoria.add(nueva_categoria);
                estado.add(nuevo_estado);
                System.out.println("================================");
                System.out.println("Recordatorio creado correctamente");
                System.out.println("================================");
            }
            //----------------------------//
            //--|editar_el_recordatorio|--//
            //----------------------------//
            else if (opcion == 2) {
                System.out.print("Ingrese el numero del recordatorio: ");
                int editar = entrada.nextInt();
                entrada.nextLine();
                if (editar <= cantidad && editar > 0) {
                    int indice = editar - 1;
                    System.out.print("Nuevo titulo: ");
                    titulo.set(indice, entrada.nextLine());
                    System.out.print("Nueva descripcion: ");
                    descripcion.set(indice, entrada.nextLine());
                    System.out.print("Nueva fecha: ");
                    fecha.set(indice, entrada.nextLine());
                    System.out.print("Nueva prioridad: ");
                    prioridad.set(indice, entrada.nextLine());
                    System.out.println("================================");
                    System.out.println("Recordatorio editado");
                    System.out.println("================================");
                } else {
                    System.out.println("Recordatorio no encontrado");
                }
            }
            //------------------------------//
            //--|eliminar_el_recordatorio|--//
            //------------------------------//
            else if (opcion == 3) {
                System.out.print("Ingrese el numero del recordatorio: ");
                int eliminar = entrada.nextInt();
                entrada.nextLine();
                if (eliminar <= cantidad && eliminar > 0) {
                    int indice = eliminar - 1;
                    titulo.remove(indice);
                    descripcion.remove(indice);
                    fecha.remove(indice);
                    hora.remove(indice);
                    prioridad.remove(indice);
                    categoria.remove(indice);
                    estado.remove(indice);
                    cantidad--;
                    System.out.println("================================");
                    System.out.println("Recordatorio eliminado");
                    System.out.println("================================");
                } else {
                    System.out.println("Recordatorio no encontrado");
                }
            }
            //----------------------------//
            //--|marcar_como_completado|--//
            //----------------------------//
            else if (opcion == 4) {
                System.out.print("Ingrese el numero del recordatorio: ");
                int completar = entrada.nextInt();
                entrada.nextLine();
                if (completar <= cantidad && completar > 0) {
                    int indice = completar - 1;
                    estado.set(indice, "Completado");
                    System.out.println("================================");
                    System.out.println("Recordatorio completado");
                    System.out.println("================================");
                } else {
                    System.out.println("Recordatorio no encontrado");
                }
            }
            //--------------------------------//
            //--|mostrar_como_recordatorios|--//
            //--------------------------------//
            else if (opcion == 5) {
                if (cantidad == 0) {
                    System.out.println("No hay recordatorios");
                } else {
                    for (int j = 0; j < cantidad; j++) {
                        System.out.println("===============================");
                        System.out.println("Recordatorio # " + (j + 1));
                        System.out.println("Titulo: " + titulo.get(j));
                        System.out.println("Descripcion: " + descripcion.get(j));
                        System.out.println("Fecha: " + fecha.get(j));
                        System.out.println("Hora: " + hora.get(j));
                        System.out.println("Prioridad: " + prioridad.get(j));
                        System.out.println("Categoria: " + categoria.get(j));
                        System.out.println("Estado: " + estado.get(j));
                        System.out.println("===============================");
                    }
                }
            }
            //------------------------------//
            //--|salir_del_menu_principal|--//
            //------------------------------//
            else if (opcion == 6) {
                System.out.println("Saliendo del sistema...");
            }
            else {
                System.out.println("Opcion invalida");
            }
        }
        entrada.close();
    }
}

File: bloc_de_recordatorios.py
opcion = 0
cantidad = 0
titulo = []
descripcion = []
fecha = []
hora = []
prioridad = []
categoria = []
estado = []
#---------------------------------------------#
#--|menu_principal_de_bloc_de_recordatorios|--#
#---------------------------------------------#
while opcion != 6:
    print("===============================")
    print("   BLOC DE RECORDATORIOS")
    print("===============================")
    print("1. Crear recordatorio")
    print("2. Editar recordatorio")
    print("3. Eliminar recordatorio")
    print("4. Marcar como completado")
    print("5. Mostrar recordatorios")
    print("6. Salir")
    print("===============================")
    opcion = int(input("Seleccione una opcion: "))
    #---------------------------#
    #--|crear_el_recordatorio|--#
    #---------------------------#
    if opcion == 1:
        cantidad += 1
        nuevo_titulo = input("Ingrese el titulo: ")
        nueva_descripcion = input("Ingrese la descripcion: ")
        nueva_fecha = input("Ingrese la fecha: ")
        nueva_hora = input("Ingrese la hora: ")
        print("Alta - Media - Baja")
        nueva_prioridad = input("Ingrese la prioridad: ")
        print("Trabajo - Estudio - Personal - Compras")
        nueva_categoria = input("Ingrese la categoria: ")
        nuevo_estado = "Pendiente"
        titulo.append(nuevo_titulo)
        descripcion.append(nueva_descripcion)
        fecha.append(nueva_fecha)
        hora.append(nueva_hora)
        prioridad.append(nueva_prioridad)
        categoria.append(nueva_categoria)
        estado.append(nuevo_estado)
        print("================================")
        print("Recordatorio creado correctamente")
        print("================================")
    #----------------------------#
    #--|editar_el_recordatorio|--#
    #----------------------------#
    elif opcion == 2:
        editar = int(input("Ingrese el numero del recordatorio: "))
        if editar <= cantidad and editar > 0:
            indice = editar - 1
            titulo[indice] = input("Nuevo titulo: ")
            descripcion[indice] = input("Nueva descripcion: ")
            fecha[indice] = input("Nueva fecha: ")
            prioridad[indice] = input("Nueva prioridad: ")
            print("================================")
            print("Recordatorio editado")
            print("================================")
        else:
            print("Recordatorio no encontrado")
    #------------------------------#
    #--|eliminar_el_recordatorio|--#
    #------------------------------#
    elif opcion == 3:
        eliminar = int(input("Ingrese el numero del recordatorio: "))
        if eliminar <= cantidad and eliminar > 0:
            indice = eliminar - 1
            titulo.pop(indice)
            descripcion.pop(indice)
            fecha.pop(indice)
            hora.pop(indice)
            prioridad.pop(indice)
            categoria.pop(indice)
            estado.pop(indice)
            cantidad -= 1
            print("================================")
            print("Recordatorio eliminado")
            print("================================")
        else:
            print("Recordatorio no encontrado")
    #----------------------------#
    #--|marcar_como_completado|--#
    #----------------------------#
    elif opcion == 4:
        completar = int(input("Ingrese el numero del recordatorio: "))
        if completar <= cantidad and completar > 0:
            indice = completar - 1
            estado[indice] = "Completado"
            print("================================")
            print("Recordatorio completado")
            print("================================")
        else:
            print("Recordatorio no encontrado")
    #--------------------------------#
    #--|mostrar_como_recordatorios|--#
    #--------------------------------#
    elif opcion == 5:
        if cantidad == 0:
            print("No hay recordatorios")
        else:
            for j in range(cantidad):
                print("===============================")
                print("Recordatorio #", j + 1)
                print("Titulo:", titulo[j])
                print("Descripcion:", descripcion[j])
                print("Fecha:", fecha[j])
                print("Hora:", hora[j])
                print("Prioridad:", prioridad[j])
                print("Categoria:", categoria[j])
                print("Estado:", estado[j])
                print("===============================")
    #------------------------------#
    #--|salir_del_menu_principal|--#
    #------------------------------#
    elif opcion == 6:
        print("Saliendo del sistema...")
    else:
        print("Opcion invalida")